### PR TITLE
Refactor model

### PR DESCRIPTION
**Pull request by oleschl**

### CODE REVIEW

File: packages/RichTextEditing-Core.package/RichTextEditor.class/class/paragraphCharacter.st [hunk 1]
```diff
@@ -1,4 +1,4 @@
 constants
 paragraphCharacter
-
+
```
Comment (frcroth) [inline packages/RichTextEditing-Core.package/RichTextEditor.class/class/paragraphCharacter.st]:
```suggestion

```

File: packages/RichTextEditing-Core.package/RichTextEditorModel.class/instance/addAttribute..st [hunk 2]
```diff
@@ -2,7 +2,7 @@ text structure
...
 	self setTextFromView.
-	self bufferDocument
-		addAttribute: aTextAttribute
-		from: self selectionStart
-		to: self selectionStop
\ No newline at end of file
+	self bufferDocument 
+			addAttribute: aTextAttribute 
+			from: self selectionStart 
+			to: self selectionStop
```
Comment (frcroth) [inline packages/RichTextEditing-Core.package/RichTextEditorModel.class/instance/addAttribute..st]:
```suggestion
	self bufferDocument
		addAttribute: aTextAttribute
		from: self selectionStart
		to: self selectionStop
```

File: packages/RichTextEditing-Core.package/RichTextEditorModel.class/instance/getStructureSelectionAt..st [hunk 3]
```diff
@@ -1,8 +1,5 @@
...
 getStructureSelectionAt: anIndex
 
-	| structures |
-	
-	self isIntervalEmpty ifTrue: [structures := self currentStructures]
-									ifFalse: [structures := self selectedStructures].
-	^ structures includes: (self textStructureSymbols at: anIndex ifAbsent: false)
\ No newline at end of file
+	^ selectedStructures includes: (self textStructureSymbols at: anIndex ifAbsent: [false])
+
```
Comment (frcroth) [inline packages/RichTextEditing-Core.package/RichTextEditorModel.class/instance/getStructureSelectionAt..st]:
```suggestion
```

File: packages/RichTextEditing-Core.package/RichTextEditorModel.class/instance/removeTextAttribute..st [hunk 4]
```diff
@@ -1,8 +1,7 @@
 text structure
 removeTextAttribute: aTextAttribute
-
+	
 	self bufferDocument
 		removeTextAttribute: aTextAttribute
 		from: self selectionStart
-		to: self selectionStop.
-	self selectionChanged
```
Comment (frcroth) [inline packages/RichTextEditing-Core.package/RichTextEditorModel.class/instance/removeTextAttribute..st:8]: Why isn't this needed anymore

File: packages/RichTextEditing-Core.package/RichTextEditorModel.class/instance/removeTextAttribute..st [hunk 5]
```diff
@@ -1,8 +1,7 @@
 text structure
 removeTextAttribute: aTextAttribute
-
+
```
Comment (frcroth) [inline packages/RichTextEditing-Core.package/RichTextEditorModel.class/instance/removeTextAttribute..st]:
```suggestion

```

File: packages/RichTextEditing-Core.package/RichTextEditorModel.class/instance/selectionChanged.st [hunk 6]
```diff
@@ -1,6 +1,6 @@
 change
 selectionChanged
 
-	self selectedStructures: (self textStructureSymbols select:
-		[:structure | self hasStructure: structure]) asSet.
+	self isIntervalEmpty ifFalse: [self selectedStructures: (self textStructureSymbols select:
```
Comment (frcroth) [inline packages/RichTextEditing-Core.package/RichTextEditorModel.class/instance/selectionChanged.st]: Rename isIntervalEmpty

File: packages/RichTextEditing-Core.package/RichTextEditor.class/class/paragraphCharacter.st [hunk 7]
```diff
@@ -1,4 +1,3 @@
 constants
 paragraphCharacter
-	
-	^ Character value: 182
\ No newline at end of file
+	^ Character value: 182
```
Comment (frcroth) [inline packages/RichTextEditing-Core.package/RichTextEditor.class/class/paragraphCharacter.st:4]:
```suggestion

	^ Character value: 182
```

File: packages/RichTextEditing-Core.package/RichTextEditorModel.class/instance/precedingStructuresChanged.st [hunk 8]
```diff
@@ -1,6 +1,6 @@
 as yet unclassified
```
Comment (sirkrypt0) [inline packages/RichTextEditing-Core.package/RichTextEditorModel.class/instance/precedingStructuresChanged.st]: classify

File: packages/RichTextEditing-Core.package/RichTextEditorModel.class/instance/precedingStructuresChanged.st [hunk 9]
```diff
@@ -1,6 +1,6 @@
...
 precedingStructuresChanged
 
-	self precedingStructures: (self textStructureSymbols select:
+	self selectedStructures: (self textStructureSymbols select:
 		[:structure | self hasStructure: structure between: self selectionStart to: self selectionStart]) asSet.
```
Comment (sirkrypt0) [inline packages/RichTextEditing-Core.package/RichTextEditorModel.class/instance/precedingStructuresChanged.st]:
```suggestion
	self selectedStructures: (self textStructureSymbols select: [:structure |
	    self hasStructure: structure between: self selectionStart to: self selectionStart]) asSet.
```

File: packages/RichTextEditing-Core.package/RichTextEditorModel.class/instance/selectionChanged.st [hunk 10]
```diff
@@ -1,6 +1,6 @@
 change
 selectionChanged
 
-	self selectedStructures: (self textStructureSymbols select:
-		[:structure | self hasStructure: structure]) asSet.
+	self isSelectionEmpty ifFalse: [self selectedStructures: (self textStructureSymbols select:
+		[:structure | self hasStructure: structure]) asSet].
```
Comment (sirkrypt0) [inline packages/RichTextEditing-Core.package/RichTextEditorModel.class/instance/selectionChanged.st]:
```suggestion
	self isSelectionEmpty ifFalse: [self selectedStructures: (self textStructureSymbols select: [:structure |
	    self hasStructure: structure]) asSet].
```

File: packages/RichTextEditing-Core.package/RichTextEditorModel.class/instance/selectionChanged.st [hunk 11]
```diff
@@ -1,6 +1,6 @@
 change
```
Comment (sirkrypt0) [inline packages/RichTextEditing-Core.package/RichTextEditorModel.class/instance/selectionChanged.st:1]: seems a weird category name

**Review by sirkrypt0 — CHANGES_REQUESTED** [review]

**Pull request merged**